airbyte-workers/src/main/java/io/airbyte/workers/helper/ResourceRequirementsUtils.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.helper;
+
+import com.google.common.base.Preconditions;
+import io.airbyte.config.ActorDefinitionResourceRequirements;
+import io.airbyte.config.ActorType;
+import io.airbyte.config.JobTypeResourceLimit;
+import io.airbyte.config.JobTypeResourceLimit.JobType;
+import io.airbyte.config.ResourceRequirements;
+import io.airbyte.config.StandardSyncInput;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ResourceRequirementsUtils {
+
+  /**
+   * Given an actor type and job type, returns resource requirements, if present. Otherwise, null.
+   * Sync-level resource requirements, take precedence over connector definition level ones.
+   *
+   * @param input - sync input
+   * @param actorType - type of actor to extract resource requirements for
+   * @param jobType - toe of job to extract resource requirements for
+   * @return resource requirements, if present, otherwise null.
+   */
+  public static ResourceRequirements getResourceRequirements(final StandardSyncInput input, final ActorType actorType, final JobType jobType) {
+    final ActorDefinitionResourceRequirements actorDefResourceReqs = getResourceRequirementsForActorType(input, actorType);
+    if (input.getResourceRequirements() != null) {
```

Comment:
One potential issue I could see with this is if the `resourceRequirements` on the connection is set to a non-null empty value like `{}`. In this case, the currently logic you have here would use the connection-level resource requirements since they are not null, but they would be empty. 

This is currently the value set in most of the connection records, so this logic would cause connector-definition level resource requirements to always be ignored:
<img width="1720" alt="Screen Shot 2022-02-22 at 11 27 07 AM" src="https://user-images.githubusercontent.com/22731524/155204568-0213ad45-192a-4c4c-a258-1cd373ac3709.png">

This is why I wrote the logic [here](https://github.com/airbytehq/airbyte/blob/93b34e28d310be3b24fe3f7c74253fda61730ba9/airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java#L102-L122) to look at each individual resource requirement value in the connection record, and only use the worker default if a specific value was not set.

Maybe this should just be changed from an `is null` check to an `is null or is empty` check?

Reply:
wrote a merge function for `ResourceRequirements`
```
 /**
   * Given a list of resource requirements, merges them together. Earlier reqs override later ones.
   *
   * @param resourceReqs - list of resource request to merge
   * @return merged resource req
   */
  public static ResourceRequirements mergeResourceRequirements(final ResourceRequirements... resourceReqs) {
    final ResourceRequirements outputReqs = new ResourceRequirements();
    final List<ResourceRequirements> reversed = new ArrayList<>(List.of(resourceReqs));
    Collections.reverse(reversed);

    for (final ResourceRequirements resourceReq : reversed) {
      if (resourceReq == null) {
        continue;
      }

      if (resourceReq.getCpuRequest() != null) {
        outputReqs.setCpuRequest(resourceReq.getCpuRequest());
      }
      if (resourceReq.getCpuLimit() != null) {
        outputReqs.setCpuLimit(resourceReq.getCpuLimit());
      }
      if (resourceReq.getMemoryRequest() != null) {
        outputReqs.setMemoryRequest(resourceReq.getMemoryRequest());
      }
      if (resourceReq.getMemoryLimit() != null) {
        outputReqs.setMemoryLimit(resourceReq.getMemoryLimit());
      }
    }
    return outputReqs;
  }
```